python/hsfs/core/arrow_flight_client.py

```diff
@@ -0,0 +1,371 @@
+#
+#   Copyright 2023 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import json
+import base64
+import warnings
+import pyarrow
+import pyarrow.flight
+from pyarrow.flight import FlightServerError
+from hsfs import client
+from hsfs import feature_group
+from hsfs.client.exceptions import FeatureStoreException
+from hsfs.core.variable_api import VariableApi
+
+
+_arrow_flight_instance = None
+
+
+def get_instance():
+    global _arrow_flight_instance
+    if not _arrow_flight_instance:
+        _arrow_flight_instance = ArrowFlightClient()
+    return _arrow_flight_instance
+
+
+class ArrowFlightClient:
+    def __init__(self):
+        self._is_initialized = False
```

Comment:
There was an option to bypass the featureflag by setting read_option["use_flyingduck"]=True. In that case I needed to 1) know if the connection was already initialized 2) be able to initialize it if needed. Since I removed the option to bypass the feature flag, we could, theoretically remove this. However I'm thinking we might still need it in the future.